agent/grpc-external/services/resource/server.go

```diff
@@ -55,7 +63,17 @@ func (s *Server) Delete(ctx context.Context, req *pbresource.DeleteRequest) (*pb
 	return &pbresource.DeleteResponse{}, nil
 }
 
-//nolint:unparam
+// Extracts ACL token ID from incoming context
+func tokenFromContext(ctx context.Context) string {
+	// TODO(spatel): Figure out where to get the token from correctly. This is just a stand in for the PR
+	//               Should I be using QueryOptions like some of the other gRPC servers?
+	token := ctx.Value("x-token")
```

Comment:
Generally, the ACL token is passed around via a gRPC metadata field called `x-consul-token`. I don't think we should carry over the `QueryOptions` stuff here because it contains a lot of stuff that doesn't apply to this API.

Reply:
👍 Fixed